server/app.py

```diff
@@ -1593,16 +1593,19 @@ def log_rdo(type=None, contact=None, account=None, subscription=None):
 def update_next_opportunity(opps=[], invoice=None):
     if not opps:
         opps = Opportunity.list(
-            stage_name="Pledged", stripe_subscription_id=invoice["subscription"]["id"]
+            stage_name="Pledged",
+            stripe_subscription_id=invoice["subscription"]["id"],
+            asc_order=True
         )
 
-    charged_on = datetime.fromtimestamp(invoice["effective_at"]).strftime('%Y-%m-%d')
-    opp = [
-        opportunity
-        for opportunity in opps
-        if opportunity.expected_giving_date == charged_on
-    ][0]
-    app.logger.info(f'opps with giving_date today on subscription_id: {opp}')
+    next_opp = opps[0]
+    next_opp_date = datetime.strptime(next_opp.expected_giving_date, "%Y-%m-%d")
+    charged_on = datetime.fromtimestamp(invoice["effective_at"])
```

Comment:
Okay, updated charged_on to charged_on_date. And, at the point, we wouldn't hit a scenario where we didn't have invoice here because the two funcs that call this one both check for invoice before getting here.